lib/org/chromium/webidl/ParserRunner.js

```diff
@@ -0,0 +1,46 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'ParserRunner',
+  extends: 'foam.box.Runnable',
+
+  requires: ['org.chromium.webidl.PipelineMessage'],
+
+  documentation: 'Runnable Box that parsers IDLFileContents into ASTs.',
+
+  methods: [
+    function run(message) {
+      if (!this.PipelineMessage.isInstance(message))
+        return this.error(new Error("ParserRunner: Run() expects a" +
+            " PipelineMessage object!"));
+      else if (!message.idlFile || !message.parser)
+        return this.error(new Error("ParserRunner: Missing required" +
+            " properties in PipelineMessage!"));
+
+      var file = message.idlFile;
+      var parser = message.parser;
+
+      // Load appropriate parser depending on configuration.
+      var parser = foam.lookup(`org.chromium.webidl.${parser}`).create();
+      var results = parser.parseString(file.contents);
+
+      // Determine if parse of entire file was successful.
+      if (file.contents.length !== results.pos) {
```

Comment:
From my observations, a failed parse does not return `undefined` nor `null`, a `StringPS` is always returned.  Additionally, `.value` is defined even on a failed parse, except it will be an array of 0 elements. However, this would also be the case for a parse on a file that has no IDL content (for example, a file that only has preprocessor directives or comments maybe).